domain/schema/controller.go

```diff
@@ -482,3 +485,65 @@ BEGIN
     VALUES (4, 5, OLD.key, DATETIME('now'));
 END;`)
 }
+
+func upgradeInfoSchema() database.Delta {
+	return database.MakeDelta(`
+CREATE TABLE upgrade_info (
+    uuid             TEXT PRIMARY KEY,
+    previous_version TEXT NOT NULL,
+    target_version   TEXT NOT NULL,
+    created_at       TIMESTAMP NOT NULL,
+    started_at       TIMESTAMP,
+    completed_at     TIMESTAMP
+);
+
+CREATE TABLE upgrade_node_status (
+    id     INT PRIMARY KEY,
+    status TEXT NOT NULL
```

Comment:
this table could be normalized into `upgrade_info_controller_node` or am I missing something?

Reply:
Yes it could be, by replacing `upgrade_node_status_id` in `upgrade_info_controller_node` with just the `status` string

However, this approach has some advantages. Namely, the foreign key constraint will constrain the status to the enum. We can then use this table as the source of truth of valid node statuses, rather than doing so in code. This seems to be the pattern we're following with other tables (see `auth_types`, `cloud_types`)

I am still on the fence whether this is actually worth doing, as I suspect it's unlikely another status is going to come along

Reply:
Perhaps an entry completed_at and a boolean comparison against null would be better